qutebrowser/commands/runners.py

```diff
@@ -57,11 +57,19 @@ def replace_variables(win_id, arglist):
                                                     QUrl.RemovePassword)
     if '{url:pretty}' in arglist:
         pretty_url = _current_url(tabbed_browser).toString(QUrl.RemovePassword)
+    if '{clipboard}' in arglist:
+        clipboard = utils.get_clipboard()
+    if '{primary}' in arglist:
+        primary = utils.get_clipboard(selection=True)
     for arg in arglist:
         if arg == '{url}':
             args.append(url)
         elif arg == '{url:pretty}':
             args.append(pretty_url)
+        elif arg == '{clipboard}':
+            args.append(clipboard)
+        elif arg == '{primary}':
+            args.append(primary)
```

Comment:
This code starts to look a bit convoluted (too many options each defining its own variable). In my opinion it should be refactored by using a dictionary (or similar declarative approach).